src/main/java/com/streamr/client/subs/BasicSubscription.java

```diff
@@ -107,6 +107,7 @@ public void run() {
     protected void handleInOrderQueue(String publisherId) {
         cancelGroupKeyRequest(publisherId);
         ArrayDeque<StreamMessage> queue = encryptedMsgsQueues.get(publisherId);
+        getLogger().trace("Checking and handling encryption queue for publisher {}. Queue size: {}", publisherId, queue.size());
```

Comment:
Maybe only log when queue is non-empty or > 1?

Reply:
This method only gets called on `setGroupKeys`, so not awfully often. `trace` is also the finest-grained logging level, basically only used when debugging. I would keep it like this, because when I was debugging a failing test, it was useful to immediately see from the logs that 1) yes this method was visited and 2) what queue size was (also seeing it was zero was useful).